### ISSUE TITLE
REST API Product Images PUT request not saving "types"

### ISSUE THREAD

**Issue by mhenkel1** [magento/magento2#18976]
**Problem description:**
When you try to update product images via the Magento REST API, the field "Types" is not saved.
For example you have not added the type "thumbnail" during the POST request, and want to add it later on. Currently you have to DELETE the image and add it again with another POST.

**Preconditions**
Magento 2
Version: 2.2.x (tested on 2.2.5 and 2.2.6)
PHP v7.1.17
Mysql 5.7
This problem was also tested a local environment with Laravel Valet on MacOS.

**Steps to reproduce**
Create new product with sku SUN001
Add 3 different images with POST /V1/products/SUN001/media
Update Types value with PUT and add "images","small_image","thumbnail". Example:
```
PUT
{{base_url}}/rest/V1/products/sun-001/media/19
{
"entry": {
"id" : "19"
"media_type": "image",
"label": "image 3",
"disabled": false,
"position" : 0,
"types": [
"images",
"small_images",
"thumbnail"
]
}
}
```
4. Try now to assign types values to the images, for example:
```
PUT
{{base_url}}/rest/V1/products/sun-001/media/15
{
"entry": {
"id" : "15"
"media_type": "image",
"label": "image 3",
"disabled": false,
"position" : 0,
"types": [
"images",
"small_images",
"thumbnail"
]

}
}
```

**Expected result**
Types values have been updated after PUT request.

**Actual result**
Types values are not saved to the DB.

**Comment by magento-engcom-team** [comment]
Hi @mhenkel1. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@mhenkel1 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by mhenkel1** [comment]
> Hi @mhenkel1. Thank you for your report.
> To help us process this issue please make sure that you provided the following information:
> 
> * [x]  Summary of the issue
> * [x]  Information on your environment
> * [x]  Steps to reproduce
> * [x]  Expected and actual results

> @mhenkel1 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
> * [x]  Already done on local testing instance, not gonna do it again.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96045`, `MAGETWO-96046` were created

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by vasilii-b** [comment]
Hi @mhenkel1,
Unfortunately I didn't reproduce the issue on both `2.2-develop` an `2.3-develop`. Taking your code as example will result into a true result - issue reproduced.
```JSON
{
"entry": {
"id" : "15"
"media_type": "image",
"label": "image 3",
"disabled": false,
"position" : 0,
"types": [
"images",
"small_images",
"thumbnail"
]

}
}
```
But, your PUT request is wrong. The problem is in type naming. Why do you add `s` at the end ? 🤔 

Here is a valid PUT request that succeeds:
```JSON
{
"entry": {
	"id" : 3,
	"media_type": "image",
	"label": "Product Image",
	"disabled": false,
	"position" : 100,
	"types": [
		"image",
		"small_image",
		"thumbnail",
		"swatch_image"
	]
}
}
```

Can you give it a try ?

**Comment by vasilii-b** [comment]
Closing the issue until new details are provided. Thank you!
